web-common/src/features/dashboards/workspace/Dashboard.svelte

```diff
@@ -112,8 +115,41 @@
 
   $: visibleMeasureNames = $visibleMeasures.map(({ name }) => name ?? "");
 
-  const urlThemeName: Writable<string | null> = writable(null);
-  $: urlThemeName.set($page.url.searchParams.get("theme"));
+  const urlThemeName = writable<string | null>(null);
+
+  $: embedThemeStore = isEmbedded ? getEmbedThemeStoreInstance() : null;
+  $: embedThemeValue = embedThemeStore ? get(embedThemeStore) : null;
+
+  $: {
+    if (isEmbedded && embedThemeValue !== undefined) {
+      const resolvedTheme = resolveEmbedTheme(isEmbedded, embedThemeValue);
+      urlThemeName.set(resolvedTheme);
+    } else {
+      urlThemeName.set($page.url.searchParams.get("theme"));
+    }
+  }
+
+  let unsubscribeTheme: (() => void) | null = null;
+  $: if (isEmbedded && embedThemeStore) {
```

Comment:
Is all this complexity needed? `embedThemeStore` doesnt really change since we init it based on embedId.

How about setting the `theme` param instead? All of this and canvas code is not needed anymore.